com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -366,7 +366,19 @@ internal Vector3 Scale
         /// </summary>
         protected NetworkManager m_CachedNetworkManager;
 
-        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkState = new NetworkVariable<NetworkTransformState>(new NetworkTransformState());
+        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkStateServer = new NetworkVariable<NetworkTransformState>(new NetworkTransformState(), NetworkVariableReadPermission.Everyone, NetworkVariableWritePermission.Server);
+        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkStateOwner = new NetworkVariable<NetworkTransformState>(new NetworkTransformState(), NetworkVariableReadPermission.Everyone, NetworkVariableWritePermission.Owner);
+
+
+        internal NetworkVariable<NetworkTransformState> GetReplicatedNetworkState()
```

Comment:
nit: Consider a getter property instead?

Reply:
Getter properties with logic are more expensive than methods?

Reply:
No, they're identical (and even if they weren't, this would be firmly in the micro-optimization territory). Properties ARE methods, and access to a property getter will be a callvirt, just like access to a method.

```
    {
        int result = app.MyProp;
        int res2 = app.MyMethod();
    }

    public int MyProp { get { /* ... */ } }

    public int MyMethod() { /* ... */ }
```

```
  // [17 9 - 17 33]
    IL_0015: ldloc.0      // app
    IL_0016: callvirt     instance int32 HelloWorld::get_MyProp()
    IL_001b: stloc.1      // result

    // [18 9 - 18 35]
    IL_001c: ldloc.0      // app
    IL_001d: callvirt     instance int32 HelloWorld::MyMethod()
    IL_0022: stloc.2      // res2

  .method public hidebysig specialname instance int32
    get_MyProp() cil managed

  .method public hidebysig instance int32
    MyMethod() cil managed
```

Perhaps you're thinking of the general guidance that "getters shouldn't be used to hide mutations or heavy calculations"?

Reply:
Yes, that is what I was thinking of...
Will change this to a getter.  👍